packages/machine-extractor/src/__tests__/parseResult.test.ts

```diff
@@ -130,109 +132,518 @@ describe('MachineParseResult', () => {
+        a: {
+          exit: [assign({count: ctx => ctx.count + 1, another: function (ctx, e) {
+            return ctx.whatever + e.anotherWhatever
+          }})],
+          on: {
+            FOO: {
+              target: 'b',
+              actions: [assign({a: 0, b: 'b', c: true, d: [1,2,3], e: {e1: 'whatever'}})]
             }
           },
-          b: {
-            entry: [${assignActions}]
-          }
-        }
-      })
-    `;
-    const result = extractMachinesFromFile(config);
+        },
+        b: {},
+      },
+    });
+  `);
     const machine = result!.machines[0];
+    const config = machine?.toConfig();
 
-    it('should extract entry assignment', () => {
-      expect(machine?.toConfig({ stringifyInlineImplementations: true })?.entry)
-        .toMatchInlineSnapshot(`
-        [
-          "assign({count: 1})",
-          "assign((ctx, e) => {
+    expect(config!.entry!.map((entry) => pick(entry, ['name', 'assignment'])))
+      .toMatchInlineSnapshot(`
+      [
+        {
+          "assignment": {
+            "inlineImplementation": {
+              "type": "expression",
+              "value": "(ctx, e) => {
                 const val = e.data;
                 return {
-                  count: val + ctx.count
-                }
-              })",
-          "assign({count: ctx => ctx.count + 1})",
-        ]
-      `);
-      expect(
-        machine?.toConfig({ stringifyInlineImplementations: true })?.states?.b
-          .entry,
-      ).toMatchInlineSnapshot(`
-        [
-          "assign({count: 1})",
-          "assign((ctx, e) => {
+                  count: val + ctx.count,
+                };
+              }",
+            },
+          },
+          "name": "xstate.assign",
+        },
+        {
+          "assignment": {
+            "inlineImplementation": {
+              "type": "expression",
+              "value": "function (ctx, e) {
                 const val = e.data;
                 return {
-                  count: val + ctx.count
-                }
-              })",
-          "assign({count: ctx => ctx.count + 1})",
-        ]
-      `);
+                  count: val + ctx.count,
+                };
+              }",
+            },
+          },
+          "name": "xstate.assign",
+        },
+      ]
+    `);
+
+    expect(pick(config?.states.a.exit, ['name', 'assignment']))
+      .toMatchInlineSnapshot(`
+      {
+        "assignment": {
+          "another": {
+            "type": "expression",
+            "value": "function (ctx, e) {
+                  return ctx.whatever + e.anotherWhatever
+                }",
+          },
+          "count": {
+            "type": "expression",
+            "value": "ctx => ctx.count + 1",
+          },
+        },
+        "name": "xstate.assign",
+      }
+    `);
+
+    ['a', 'b', 'c', 'd', 'e'].forEach((prop) => {
+      expect(config?.states?.a?.on?.FOO?.actions.assignment).toHaveProperty(
+        prop,
+      );
     });
```

Comment:
this bit feels redundant if you are testing all of those in just the next `expect` call